turbinia/workers/analysis/auth.py

```diff
@@ -0,0 +1,619 @@
+    if not username or username.lower() == 'nan':
+      return username
+    return f'{domain}\\{username}'
+
+  def from_useraccount(self, useraccount: str):
+    """Split useraccount into domain and username."""
+
+    if not '\\' in useraccount:
+      return '', useraccount
+
+    val = useraccount.split('\\')
+    try:
+      domain = val[0].strip()
+      username = val[1].strip()
+      return domain, username
+    except ValueError:
+      return '', username
+
+  def human_timestamp(self, timestamp: int) -> str:
+    """Convert epoch timestamp to humand readable date/time."""
+    return datetime.fromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
+
+  def get_login_session(
+      self, source_ip: str, domain: str, username: str,
+      session_id: str) -> dict:
+    """Get loging session details."""
+    login_session = {
+        'source_ip': source_ip,
+        'domain': domain,
+        'username': username,
+        'session_id': session_id,
+        'login_timestamp': 0,
+        'logout_timestamp': 0,
+        'session_duration': 0,
+    }
+
+    if self.df.empty:
+      log.debug('Source dataframe is empty')
+      return login_session
+    df = self.df
+    try:
+      df_session = df[(df['source_ip'] == source_ip)
+                      & (df['username'] == username) &
+                      (df['session_id'] == session_id)]
+      login_ts = int(
+          df_session[df_session['auth_result'] == 'success'].iloc[0]
+          ['timestamp'])
+      logout_ts = int(
+          df_session[df_session['event_type'] == 'disconnection'].iloc[0]
+          ['timestamp'])
+      session_duration = logout_ts - login_ts
+      login_session['login_timestamp'] = login_ts
+      login_session['logout_timestamp'] = logout_ts
+      login_session['session_duration'] = session_duration
+    except KeyError as exception:
+      log.error(
+          f'Session duration calculation failed due to key error:'
+          f' {exception}')
+    except IndexError as exception:
+      log.error(
+          f'Session duration calculation failed due to index error:'
+          f' {exception}')
+    finally:
+      return login_session
+
+
+class BruteForceAnalyzer(AuthAnalyzer):
+  """Analyzer for brute force authentication."""
+
+  NAME = 'bruteforce.auth.analyzer'
+  DISPLAY_NAME = 'Brute Force Analyzer'
+  DESCRIPTION = 'This analyzer identifies brute force authentication'
+
+  # The time duration before a successful login event to evalute for
+  # brute force activity.
+  BRUTE_FORCE_WINDOW = 3600
+
+  # The minimum number of failed events that must occure to be considered
+  # for brute force activity.
+  BRUTE_FORCE_MIN_FAILED_EVENT = 20
+
+  # The time duration, in seconds, between successive authentication
+  # events to be considered for brute force activity.
+  BRUTE_FORCE_NEXT_LOGIN_DELTA = 10
+
+  # The minimum duration where the attacker accessed the host after
+  # a successful brute for login.
+  BRUTE_FORCE_MIN_ACCESS_DURATION = 600
```

Comment:
Updated the comment with context. This value is a threshold for a session to be considered as interactive session. This value is currently unused. It will be used to alter the priority of analyzer output.